### PR TITLE
docs(storage-classes): add section on default `StorageClass`

### PR DESCRIPTION
**Pull request by agilgur5**
## Summary

Add a section to Storage Classes briefly describing the default StorageClass
- I was looking to add this to the docs myself, but then saw that #15575 had added this section and went through a few review cycles but was never completed and merged
  - So initially based it off that, rebasing and addressing all the feedback therein ([these](https://github.com/kubernetes/website/pull/15575#discussion_r317714805) [two](https://github.com/kubernetes/website/pull/15575#pullrequestreview-279732022) points specifically). I ended up rewriting significant chunks as part of the feedback as well as after generally reviewing the [style guide](https://kubernetes.io/docs/contribute/style/style-guide/) and style of the existing doc, trying to more closely match those.

## Details

- the PVC docs reference this and there is an administrator guide on how to change this, but no section within the central Storage Classes document
  - this has made it a bit difficult to explain how that a default StorageClass exists as the StorageClass docs themselves barely mention it
    - for new cluster admins I was teaching in particular, I wanted to make sure they were aware of the pitfall that I hit with kOps in https://github.com/kubernetes/kubernetes/issues/34549#issuecomment-643501992, so the note about more than one default was particularly relevant (as well as how the [implemented fix](https://github.com/kubernetes/kubernetes/pull/110559) [works](https://github.com/kubernetes/kubernetes/pull/110559/files#diff-77fd84defc818b9e6010794bf87e0eece0960dfeff196e19b7d84da8de15ddffR67))
  - with a new section, this makes it very clear and links out to other existing docs on the topic

### CODE REVIEW

**Comment by agilgur5** [comment]
just added an empty commit at the beginning of the branch to use for the squashed title, per the [PR template directions](https://github.com/kubernetes/website/blob/8ab89bff256d26fbd3029dbb44f94a753df6812c/.github/PULL_REQUEST_TEMPLATE.md?plain=1#L9)

File: content/en/docs/concepts/storage/storage-classes.md [hunk 1]
```diff
@@ -57,6 +57,17 @@ mountOptions:
...
 
+### Default StorageClass
+
+When a PVC does not specify a `storageClassName`, the default StorageClass is
+used. The cluster can only have one default StorageClass. If more than one
+default StorageClass is accidentally set, the newest default is used when the
+PVC is dynamically provisioned.
+
+For instructions on setting the default StorageClass, see
+[Change the default StorageClass](https://kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/).
```
Comment (reylejano) [inline content/en/docs/concepts/storage/storage-classes.md:68]:
```suggestion
[Change the default StorageClass](/docs/tasks/administer-cluster/change-default-storage-class/).
```

Comment (agilgur5) [reply content/en/docs/concepts/storage/storage-classes.md:68]: oh good catch, I missed changing that from the previous PR

**Comment by reylejano** [comment]
re-applying lgtm from this [comment](https://github.com/kubernetes/website/pull/41843#issuecomment-1617138752)
/lgtm

**Comment by reylejano** [comment]
thank you for addressing my suggestion
/approve

File: content/en/docs/concepts/storage/storage-classes.md [hunk 2]
```diff
@@ -57,6 +57,17 @@ mountOptions:
 volumeBindingMode: Immediate
 ```
 
+### Default StorageClass
+
+When a PVC does not specify a `storageClassName`, the default StorageClass is
+used. The cluster can only have one default StorageClass. If more than one
+default StorageClass is accidentally set, the newest default is used when the
```
Comment (chenrui333) [inline content/en/docs/concepts/storage/storage-classes.md:64]: 👋 why not just use `latest default` rather than `newest default`?

**Pull request merged**